packages/integrations/markdoc/components/TreeNode.ts

```diff
@@ -35,75 +38,86 @@ export type TreeNode =
 			children: TreeNode[];
 	  };
 
-export const ComponentNode = createComponent({
-	factory(result: any, { treeNode }: { treeNode: TreeNode }) {
-		if (treeNode.type === 'text') return render`${treeNode.content}`;
-
```

Comment:
It might look like I'm doing a lot of change here, but I'm only extracting the factory function here out so I can call itself for arrays. Disabling whitespace in diff should make it easier to review.